### ISSUE TITLE
Ackermann steering controller example not working properly

### ISSUE THREAD

**Issue by Victorvi99** [ros-controls/gz_ros2_control#438]
Hi, I am trying to use the '/gz_ros2_control/ign_ros2_control_demos/launch/ackermann_drive_example.launch.py' as an inspiration for my ackermann simulation. 

When launching the ackermann_drive_example.launch.py and teleoperating it with the keyboard (vx=0.5 and wz=0.3) I experienced that the car turns to the right in Ignition when sending positive angular velocities. However, when visualizing the odom frame in rviz published by the ackermann steering controller, the car (shown by robotmodel subscribed to robot_description) turns to the left, as expected with positive angular velocities. 

I would expect the car turning to the left also in Ignition following the robot base link convention (x: forward, y:left, z:up) in which the right hand rule says that positive angular velocities would turn to the left. 

By inspecting the test_ackermann_drive.xacro.urdf, I could see that the 'left_wheel_steering_joint' and the 'right_wheel_steering_joint' define rpy="-1.57 0 0" which turns the base_link frame -90deg around the x axis leading to x:forward, y:down, z:left. Then, the axis section within the joint is defined as xyz="0 1 0" ,which means that the steering will be around the y axis transformed which points downwards, so in that case, a positive angular velocity is to the right as happening in Ignition when launching the simulation. 

Trying to invert this behaviour to get positive angular velocities turning the car to the left, I have tried to change the 'left_wheel_steering_joint' and the 'right_wheel_steering_joint' origin section rpy="1.57 0 0" turning the base_link frame 90deg around the x axis leading to x:forward,y:up, z:right with axis="0 1 0" so that a positive angular velocity around y axis would turn to the left as expected. However, when doing that change and launching the simulation, the car turns to the left with positive angular velocities but the turning radius is heavily affected (from ~3m with the original setup to ~10m with the rpy joint change". It seems that is kind of drifting and not turning as it should.

I have also tried to maintain the original rpy rotation of the steering joints and changing the axis section xyz="0 -1 0" but it resulted in the same behaviour.

I would like to know if someone of you have experienced something similar as I could not found a solution for it. 

Let me know if you can test it and which is your output.

**Comment by christophfroehlich** [comment]
You are right, the movement is not correct, see also [our docs](https://control.ros.org/humble/doc/ros2_controllers/doc/mobile_robot_kinematics.html#ackermann-steering) for reference. I'll have a look

**Comment by christophfroehlich** [comment]
@Victorvi99 have a look at #468, I fixed the joint definition and this seems to be fine now. The "drift" is still there, I think this is a combination of sideslip of the wheels and integration errors of the odometry calculation.

**Comment by Narukara** [comment]
The left front wheel and right front wheel seem to be reversed. Additionally, the rotation direction of the front wheels also appears to be inverted.

https://github.com/ros-controls/gz_ros2_control/blob/5b7f233de637c4bbbd6a15bb5c4677b21b44e4f0/gz_ros2_control_demos/urdf/test_ackermann_drive.xacro.urdf#L68-L69

https://github.com/ros-controls/gz_ros2_control/blob/5b7f233de637c4bbbd6a15bb5c4677b21b44e4f0/gz_ros2_control_demos/urdf/test_ackermann_drive.xacro.urdf#L95-L96

https://github.com/ros-controls/gz_ros2_control/blob/5b7f233de637c4bbbd6a15bb5c4677b21b44e4f0/gz_ros2_control_demos/urdf/test_ackermann_drive.xacro.urdf#L111-L112

https://github.com/ros-controls/gz_ros2_control/blob/5b7f233de637c4bbbd6a15bb5c4677b21b44e4f0/gz_ros2_control_demos/urdf/test_ackermann_drive.xacro.urdf#L128-L129


![Image](https://github.com/user-attachments/assets/9c8aea24-9031-40d6-baa4-15c7164b5971)

**Comment by Narukara** [comment]
I must admit that I'm new to `ros2_control` and Gazebo. I'm not sure if there's an issue with the code or if I made a mistake.

Although the left front wheel and right front wheel are reversed, the Ackermann angle appears to be correct. Could this be because the joint order in `ackermann_drive_controller.yaml` is reversed? How does the `ackermann_steering_controller` determine which joint is on the left or right?

I tried modifying some code, and now the Ackermann vehicle seems to be working properly. I've attached my changes here: https://gist.github.com/Narukara/85615c9e381f49ebe1937f558744397e

**Comment by Narukara** [comment]
I found evidence in the [ros2_controllers documentation](https://control.ros.org/rolling/doc/ros2_controllers/steering_controllers_library/doc/userdoc.html#parameters). I think I'm ready to submit a PR. 😄 

![Image](https://github.com/user-attachments/assets/9e0a92c2-dfd1-4a8a-a44f-33e4823fedbf)

**Comment by ahcorde** [comment]
Thank you for the contribution @Narukara, related PR is merged.
